homeassistant/components/airthings_ble/sensor.py

```diff
@@ -0,0 +1,176 @@
+        key="pressure",
+        device_class=SensorDeviceClass.PRESSURE,
+        native_unit_of_measurement=PRESSURE_MBAR,
+        name="Pressure",
+    ),
+    "battery": SensorEntityDescription(
+        key="battery",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        name="Battery",
+    ),
+    "co2": SensorEntityDescription(
+        key="co2",
+        device_class=SensorDeviceClass.CO2,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        name="co2",
+    ),
+    "voc": SensorEntityDescription(
+        key="voc",
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_BILLION,
+        name="VOC",
+        icon="mdi:cloud",
+    ),
+    "illuminance": SensorEntityDescription(
+        key="illuminance",
+        native_unit_of_measurement=LIGHT_LUX,
+        name="Illuminance",
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Airthings BLE sensors."""
+    is_metric = hass.config.units.is_metric
+
+    coordinator: DataUpdateCoordinator[AirthingsDevice] = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+
+    # we need to change some units
+    if not is_metric:
+        for key, val in SENSORS.items():
+            if val.native_unit_of_measurement is not VOLUME_BECQUEREL:
+                continue
+            SENSORS[key].native_unit_of_measurement = VOLUME_PICOCURIE
+
+    entities = []
+    _LOGGER.debug("got sensors: %s", coordinator.data.sensors.keys())
+    for sensor_type, sensor_value in coordinator.data.sensors.items():
+        if sensor_type not in SENSORS:
+            _LOGGER.debug(
+                "Unknown sensor type detected: %s, %s",
+                sensor_type,
+                sensor_value,
+            )
+            continue
+        entities.append(
+            AirthingsSensor(coordinator, coordinator.data, SENSORS[sensor_type])
+        )
+
+    async_add_entities(entities)
+
+
+class AirthingsSensor(
+    CoordinatorEntity[DataUpdateCoordinator[AirthingsDevice]], SensorEntity
+):
+    """Airthings BLE sensors for the device."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        airthings_device: AirthingsDevice,
+        entity_description: SensorEntityDescription,
+    ) -> None:
+        """Populate the airthings entity with relevant data."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+
+        name = f"{airthings_device.name} {airthings_device.identifier}"
+
+        self._attr_unique_id = f"{name}_{entity_description.key}"
+
+        self._id = airthings_device.address
+        self._attr_device_info = DeviceInfo(
+            identifiers={
```

Comment:
We can use connections here instead.

`CONNECTION_BLUETOOTH`

Reply:
What do you mean? Saw that it has a connection field, but should I replace another field with that?

Reply:
Just replace `DOMAIN` with `CONNECTION_BLUETOOTH`. Just so that HA knows `airthings_device.address` is the address of a bluetooth device.

Reply:
Oh, got it. What functional benefits does this have? :)

Reply:
If other integrations also talk to the device or the a device tracker integration is tracking it they can link the device